tensorflow_datasets/core/as_dataframe.py

```diff
@@ -140,14 +140,39 @@ def _repr_html_(self) -> str:
     return self.__styler._repr_html_()  # pylint: disable=protected-access
 
 
+def _flatten_with_path(structure: TreeDict):
+  """Convert a TreeDict into a flat list of paths and their values
+
+  A path is a list of keys to get to the value
+  Yields the result elements instead of creating a list
+  """
+  if isinstance(structure, collections.Mapping):
+    for key in sorted(structure):
+      for sub_path, sub_value in _flatten_with_path(structure[key]):
+        yield [key] + sub_path, sub_value
+  else:
+    yield [], structure
+
+def _flatten(structure: TreeDict):
```

Comment:
Could `tf.nest.flatten` be used instead ?

Reply:
I guess yes. Although I'd argue that this is faster due to handling less cases and yielding instead of manifesting the whole list. I don't mind either way, your decision

Reply:
I haven't benchmarked, but  as `tf.nest.flatten`, `tree.flatten` are implemented in C/C++, I would still expect them to be faster than Python-based implementations.

For consistency, let's not try to have duplicated API. It will make the migration simpler if we replace all `tf.nest.Xyz` by `tree.Xyz`.